tests/unit_tests/grid/test_points_functions.py

```diff
@@ -0,0 +1,413 @@
+    np.testing.assert_array_almost_equal(corner_points, expected_corner_points)
+
+
+def test_corner_points_cell_split_coordinate_lines(faulted_grid):
+    # Arrange
+    cell = (0, 0, 0)
+    expected_corner_points = np.array([[[[1000., 2000., 3000.], [1100., 2000., 3000.]],
+                                        [[1000., 2100., 3000.], [1100., 2100., 3001.]]],
+                                       [[[1000., 2000., 3020.], [1100., 2000., 3020.]],
+                                        [[1000., 2100., 3020.], [1100., 2100., 3021.]]]])
+    # Act
+    corner_points = pf.corner_points(faulted_grid, cell_kji0 = cell)
+
+    # Assert
+    np.testing.assert_array_almost_equal(corner_points, expected_corner_points)
+
+
+def test_centre_point_deafult(basic_regular_grid):
+    # Arrange
+    expected_centre_points = np.array([[[[50.0, 25.0, 10.0], [150.0, 25.0, 10.0]],
+                                        [[50.0, 75.0, 10.0], [150.0, 75.0, 10.0]]],
+                                       [[[50.0, 25.0, 30.0], [150.0, 25.0, 30.0]],
+                                        [[50.0, 75.0, 30.0], [150.0, 75.0, 30.0]]]])
+    # Act
+    centre_points = pf.centre_point(basic_regular_grid)
+
+    # Assert
+    np.testing.assert_array_almost_equal(centre_points, expected_centre_points)
+
+
+def test_centre_point_k_gaps(s_bend_k_gap_grid):
+    # Act
+    centre_points = pf.centre_point(s_bend_k_gap_grid)
+
+    # Assert
+    # Large array so only checking a subset of points.
+    np.testing.assert_array_almost_equal(centre_points[2, 7, 1], np.array([26., 60., 123.28]))
+    np.testing.assert_array_almost_equal(centre_points[3, 9, 34], np.array([-0.175204, 76., 164.817584]))
+    np.testing.assert_array_almost_equal(centre_points[1, 5, 21], np.array([60.559817, 44., 148.909904]))
+    np.testing.assert_array_almost_equal(centre_points[2, 8, 29], np.array([11.227031, 68., 154.599969]))
+
+
+def test_centre_point_split_coordinate_lines(faulted_grid):
+    # Act
+    centre_points = pf.centre_point(faulted_grid)
+
+    # Assert
+    # Large array so only checking a subset of points.
+    np.testing.assert_array_almost_equal(centre_points[2, 4, 1], np.array([1150., 2450., 3045.]))
+    np.testing.assert_array_almost_equal(centre_points[0, 2, 7], np.array([1750., 2250., 3008.875]))
+    np.testing.assert_array_almost_equal(centre_points[1, 1, 4], np.array([1450., 2150., 3033.75]))
+    np.testing.assert_array_almost_equal(centre_points[2, 3, 5], np.array([1550., 2350., 3042.5]))
+
+
+# Need to check third test case
+@pytest.mark.parametrize("interpolation_fraction, expected_interpolated_point",
+                         [(np.array([0.0, 0.0, 0.0]), np.array([0.0, 0.0, 0.0])),
+                          (np.array([0.75, 0.75, 0.75]), np.array([75.0, 37.5, 15.0])),
+                          (np.array([0.75, 0.2, 0.1]), np.array([10.0, 10.0, 15.0]))])
+def test_interpolated_point(basic_regular_grid, interpolation_fraction, expected_interpolated_point):
```

Comment:
Please add a test like this using the s bend faulted grid, using a cell in the middle of the grid (not 0,0,0).